x-pack/plugins/security_solution/public/app/home/index.tsx

```diff
@@ -15,25 +15,33 @@ import { HelpMenu } from '../../common/components/help_menu';
 import { AutoSaveWarningMsg } from '../../timelines/components/timeline/auto_save_warning';
 import { UseUrlState } from '../../common/components/url_state';
 import { useWithSource } from '../../common/containers/source';
+import { useFullScreen } from '../../common/containers/use_full_screen';
+import { getPageContainerTop, hasCompactHeader } from '../../common/utils/route/helpers';
+import { useRouteSpy } from '../../common/utils/route/use_route_spy';
 import { useShowTimeline } from '../../common/utils/timeline/use_show_timeline';
 import { navTabs } from './home_navigations';
 import { useSignalIndex } from '../../detections/containers/detection_engine/alerts/use_signal_index';
+import { SecurityPageName } from '../types';
 
 const WrappedByAutoSizer = styled.div`
   height: 100%;
 `;
 WrappedByAutoSizer.displayName = 'WrappedByAutoSizer';
 
-const Main = styled.main`
-  height: 100%;
+/** the global Kibana navigation at the top of every page */
+export const globalHeaderHeightPx = 49;
+
+const Main = styled.main<{ $globalFullScreen: boolean; top: number }>`
```

Comment:
for `$globalFullScreen` and `top` as they will change whenever user toggles full screen maybe it would be a good idea to use `.attrs()` instead  https://scalablecss.com/styled-components-attrs/